File: backend/src/main/java/com/jobportal/backend/service/AdminService.java
package com.jobportal.backend.service;

import com.jobportal.backend.dto.ReportResponse;
import com.jobportal.backend.dto.UserSummary;
import com.jobportal.backend.model.Job;
import com.jobportal.backend.model.enums.JobStatus;

import java.util.List;

public interface AdminService {
    List<UserSummary> getAllUsers();

    UserSummary toggleUserActivation(Long userId, boolean active);

    void deleteUser(Long userId);

    Job updateJobStatus(Long jobId, JobStatus status);

    void deleteJob(Long jobId);

    ReportResponse generateReport();
}

File: frontend/job-portal-frontend/src/app/core/services/admin.service.ts
import { Injectable } from '@angular/core';
import { HttpClient, HttpParams } from '@angular/common/http';
import { Observable } from 'rxjs';
import { environment } from '../../../environments/environment';
import { User } from '../models/user.model';
import { Job, JobStatus } from '../models/job.model';

export interface ReportResponse {
  totalUsers: number;
  totalJobs: number;
  totalApplications: number;
}

@Injectable({ providedIn: 'root' })
export class AdminService {
  constructor(private http: HttpClient) {}

  listUsers(): Observable<User[]> {
    return this.http.get<User[]>(`${environment.apiUrl}/admin/users`);
  }

  toggleUserActivation(userId: number, active: boolean): Observable<User> {
    const params = new HttpParams().set('active', active);
    return this.http.patch<User>(`${environment.apiUrl}/admin/users/${userId}`, null, { params });
  }

  deleteUser(userId: number): Observable<void> {
    return this.http.delete<void>(`${environment.apiUrl}/admin/users/${userId}`);
  }

  updateJobStatus(jobId: number, status: JobStatus): Observable<Job> {
    const params = new HttpParams().set('status', status);
    return this.http.patch<Job>(`${environment.apiUrl}/admin/jobs/${jobId}/status`, null, { params });
  }

  deleteJob(jobId: number): Observable<void> {
    return this.http.delete<void>(`${environment.apiUrl}/admin/jobs/${jobId}`);
  }

  report(): Observable<ReportResponse> {
    return this.http.get<ReportResponse>(`${environment.apiUrl}/admin/reports`);
  }
}